### DISCUSSION TITLE
Processing of all access logs in a folder

### DISCUSSION THREAD

**Discussion by TubaApollo** [allinurl/goaccess#2549] [Q&A] [answered]
I am trying to get goaccess to process the last 60 days of access.logs. Logrotate is configured to only store the last 7 days, I could theoretically increase it to 60 days, but if I understand correctly goaccess can also process the data it already has. 

I use this command: goaccess -f access.* -a -o /path/report.html --geoip-database /path/GeoLite2-Country.mmdb --ignore-crawlers --ignore-panel=STATUS_CODES --ignore-panel=NOT_FOUND --ignore-panel=HOSTS --ignore-panel=REQUESTS --restore --persist --keep-last=60 --db-path /path/tmp/db/.

Now there are two things that irritated me:
I am pretty sure that it only processes the current access.log, not the older ones in the format: access.log-year-month-day. I guess access.* won't work, if that's the case what else should I use?
The other thing I noticed: Every time I recreate a report.html, the parsing time on the dashboard increases by one second. I don't really notice any difference in time when generating it though?

Any idea what I have misunderstood or what is wrong with the command and how I can achieve what I want? Thanks in advance! If there are similar discussions, just link to them.

**Comment by TubaApollo** [comment]
Okay it actually seems to work for multiple days, then just the question why the log parsing time gets higher and higher on the dashboard with every creation?

**Comment by allinurl** [comment]
Glad you figured out the parsing of the logs. GoAccess should support the ability to go through multiple logs, whether they're the current access.log or access.log.date* (assuming uncompressed logs). 

As for time processing, it depends on whether you're parsing the same log again and if you're using `--persist` as well. The parser will add up the seconds it takes to go through the log and increment it. You might also want to try using `--process-and-exit` to see if that helps.

**Reply by TubaApollo** [reply]
Thanks for your answer! If I remove the `--persist flag` I will probably have worse performance as it will not save newly parsed data? And if I just update the database with `--process-and-exit`, the report.html will probably need to be recreated, or am I wrong? Right now it works perfectly, with the command, but I'm kinda worried that I'll have performance issues with 60 log files.
And is it possible to show only one day for a particular block, but all 60 days for the rest? I want the static requests to show only one day's data. It is just a nice to have, so not that important.

**Comment by allinurl** [comment, marked as answer]
If your current setup is functioning well, there's no need to make changes. I anticipate no performance problems while processing those 60 files. What size are those files?

**Reply by TubaApollo** [reply]
Difficult to say as it depends on the traffic, but currently around 800kb each.

**Reply by allinurl** [reply]
It should fly through those :)

**Reply by TubaApollo** [reply]
I'll mark that as answer, i think it will work out like that, thank you so much!

**Reply by allinurl** [reply]
np. feel free to post back if you have any other questions.
